js_modules/dagster-ui/packages/ui-core/src/assets/asset-checks/AssetChecks.tsx

```diff
@@ -0,0 +1,154 @@
+import {gql, useQuery} from '@apollo/client';
+import {Body2, Box, Colors, Tag} from '@dagster-io/ui-components';
+import React from 'react';
+import {useHistory, useLocation} from 'react-router';
+
+import {LoadingSpinner} from '../../ui/Loading';
+import {useFormatDateTime} from '../../ui/useFormatDateTime';
+import {AssetKey} from '../types';
+
+import {
+  ASSET_CHECK_EXECUTION_FRAGMENT,
+  AssetCheckDetailModal,
+  MigrationRequired,
+  NoChecks,
+} from './AssetCheckDetailModal';
+import {VirtualizedAssetCheckTable} from './VirtualizedAssetCheckTable';
+import {AssetChecksQuery, AssetChecksQueryVariables} from './types/AssetChecks.types';
+
+export const AssetChecks = ({
+  lastMaterializationTimestamp,
+  lastMaterializationRunId,
+  assetKey,
+}: {
+  assetKey: AssetKey;
+  lastMaterializationTimestamp: string | undefined;
+  lastMaterializationRunId: string | undefined;
+}) => {
+  const formatDatetime = useFormatDateTime();
+  const lastMaterializationDate = React.useMemo(
+    () => (lastMaterializationTimestamp ? new Date(parseInt(lastMaterializationTimestamp)) : null),
+    [lastMaterializationTimestamp],
+  );
+  const isCurrentYear = React.useMemo(
+    () => lastMaterializationDate?.getFullYear() === new Date().getFullYear(),
```

Comment:
Stupid nit but I'm not sure it's worth memoizing this, especially because it's unstable since it depends on `new Date()`